src/system_metrics/mongoose_system_metrics_collector.erl

```diff
@@ -188,3 +189,35 @@ get_outgoing_pools() ->
     [#{report_name => outgoing_pools,
        key => type,
        value => Type} || {Type, _, _, _, _} <- OutgoingPools].
+
+get_xmpp_stanzas_count() ->
+    Hosts = ejabberd_config:get_global_option(hosts),
```

Comment:
It's possible to configure all metrics to be `global`. In this case looking at `hosts` will return nothing. Take a look at the `all_metrics_are_global` configuration option.

Reply:
I'm thinking getting a list of local and global values from mongoose_metrcis:get_value , filter out the errors and proceed with extracting values.

Reply:
As I mentioned in my other comment, the following trick from exometer API could be used: `exometer:get_values(['_', xmppMessageSent])`. In this case, you don't have to bother what is the configuration, what are the hosts or if there were errors reading a metric.

Reply:
Sure, sorry for the confusion. I was trying to address the comments one by one reading from the top and only came later to the one you mention. But totally agree, will get start working on this approach.